src/static_julia.jl

```diff
@@ -156,36 +162,43 @@ function julia_flags(optimize, debug, cc_flags)
 end
 
 function build_julia_cmd(
-        sysimage, compile, cpu_target, optimize, debug, inline, check_bounds,
-        math_mode, depwarn, startupfile = false
+        sysimage, precompiled, compilecache, home, startup_file, handle_signals,
+        compile, cpu_target, optimize, debug, inline, check_bounds, math_mode, depwarn
     )
+    # TODO: these may just be temporary workarounds, see: https://github.com/JuliaLang/PackageCompiler.jl/issues/47
+    precompiled == nothing && cpu_target == nothing || (precompiled = "no")
+    compilecache == nothing && (compilecache = "no")
```

Comment:
hmm, i could be misunderstanding, but the first one doesn't feel correct to me. 
If the user passes `precompiled = "yes"`, won't this set precompiled to "no"?

I think maybe you want parentheses around the second two phrases?
```
 precompiled == nothing && (cpu_target == nothing || (precompiled = "no"))
```
But honestly that's still hard to parse. perhaps this can simply be an `if`-statement? I think this is what you meant?:
```
if (precompiled == nothing && cpu_target != nothing) precompiled = "no" end
# or this:
(precompiled == nothing && cpu_target != nothing) && (precompiled = "no")
```

Reply:
In Julia (like in Bash) `&&` is passed through only if the previous expression returns `true`, while `||` is passed through only if the previous expression returns `false`. So if `precompiled = "yes"`, then `precompiled == nothing` returns `false`, and therefore `&&` is not passed through.

Reply:
:) So I think everything you just said is correct, and yet, behold:
```
julia> false && true || println("HI")
HI
```
I think the nuance is here:
> while `||` is passed through only if the previous expression returns `false`.

In this case, the _previous expression_ is actually _the truth-value on the left-hand side of the ||_. I would imagine it like this:
```
`precompiled == nothing` --> <evaluate expression ... okay, it's false!> --> `false` --> `&& (cpu_target == nothing)` --> <okay, skip the right-hand side, cause this already evaluates to false --> `false` --> `|| (precompiled = "no")` --> <now we need to evaluate the right-hand side again because the left-hand side was false, so the truthiness of the whole expression depends on the right-hand side> --> (precompiled = "no") --> <returns `nothing`>.
``` 

That's why I suggested, if you want to keep it exactly as you currently have it, you need to put parens around the entire right-hand side _of `&&`_, so that it can skip the whole thing:
```
julia> false && (true || println("hello?"))
>> false
```
But i still think an `if`-statement might be better, since, clearly, multiple short-circuiting statements can be confusing and hard to read. :)

Reply:
(In bash as well):
```
$ false && true || echo "HI"
HI
$ false && (true || echo "helloooooooooo")
$
```

Reply:
You're perfectly right of course :) I'll fix that, thanks!

Reply:
This is pretty straightforward:
```
precompiled == nothing && cpu_target != nothing && (precompiled = "no")
```